kroxylicious-schema-tools/kroxylicious-maven-plugin/src/main/java/io/kroxylicious/CompileSchemaMojo.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+import io.kroxylicious.tools.schema.compiler.FatalException;
+import io.kroxylicious.tools.schema.compiler.SchemaCompiler;
+
+@Mojo(name = "compile-schema", defaultPhase = LifecyclePhase.GENERATE_SOURCES)
+public class CompileSchemaMojo extends AbstractMojo {
+
+    @Parameter(defaultValue = "${project}")
+    private MavenProject project;
+
+    /**
+     * The input file or directory to be used for generating sources
+     *
+     */
+    @Parameter(property = "kroxylicious.schema.source", defaultValue = "${basedir}/src/main/schema")
+    File source;
```

Comment:
I don't think Maven supports `Path` in the conventions it applies for injecting configuration into mojos. 

Right now the tool looks for schema in `src/main/schema` which I think is enough to avoid the immediate need for globbing. I'd like the tool to be no more sophisticated than we actually need, especially while it's not something we're requiring filter authors to use.

Reply:
[This doc](https://maven.apache.org/guides/mini/guide-configuring-plugins.html#Mapping_Value_Objects) confirms that Path is not supported